File: csharp/collatz-conjecture/CollatzConjecture.cs
using System;

public static class CollatzConjecture
{
    public static int Steps(int number)
    {
        if (number <= 0)
        {
            throw new ArgumentException("The number is cannot be less than or equal with zero.");
        }
        int steps = 0;
        while (number != 1)
        {
            if (number % 2 == 0)
            {
                number = number / 2;
            }
            else
            {
                number = number * 3 + 1;
            }
            steps++;
        }
        return steps;
    }
}

File: cpp/collatz-conjecture/collatz_conjecture.cpp
#include "collatz_conjecture.h"
#include <stdexcept>

namespace collatz_conjecture
{

    int steps(int n)
    {
        if (n <= 0)
            throw std::domain_error("Number is zero or negative");

        int steps = 0;

        while (n != 1)
        {
            n = n % 2 == 0 ? n / 2 : n * 3 + 1;
            steps++;
        }

        return steps;
    }

} // namespace collatz_conjecture
